src/generate.js

```diff
@@ -42,45 +36,59 @@ module.exports = async function(req, res) {
         }
     }
 
-    const page = await global.browser.newPage();
-    if (req.body['viewportDimensions']) {
-        await page.setViewport({
-            width: Number(req.body['viewportDimensions'].split('x')[0]),
-            height: Number(req.body['viewportDimensions'].split('x')[1])
-          });
+    try {
+        const page = await global.browser.newPage();
+
+        if (req.body['viewportDimensions']) {
+            await page.setViewport({
+                width: Number(req.body['viewportDimensions'].split('x')[0]),
+                height: Number(req.body['viewportDimensions'].split('x')[1])
+            });
+        }
+
+        if (req.body.url) {
+            let response = await page.goto(req.body.url, {waitUntil: req.body['waitUntil'] ?? 'networkidle0'});
+            if (response.status() >= 400) {
+                return res.status(400).json({success: false, errors: [{msg: `Cannot access URL ${req.body.url}, the server returned an error ${response.status()}`}]})
```

Comment:
QUEST: on retourne pas du 500 aussi? c'est une erreur potentiellement client (si <500) mais côté serveur pour le client final ie l'utilisateur non ? c'est un peu poupées russes ce truc

Reply:
Ce qui risque d'arriver le plus souvent c'est une 404 not found en retour du page.goto => pour moi c'est bien une erreur client qui a fournit une req.body.url pourrie